### ISSUE TITLE
Prematurely lint against const numbers that would trigger `overflowing_literals` on platforms with smaller `usize`/`isize`s

### ISSUE THREAD

**Issue by paolobarbolini** [rust-lang/rust-clippy#13943]
### What it does

Detect const `usize`/`isize` values that correctly build for 64 bit targets but break the build with [`overflowing_literals`](https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html#overflowing-literals) on 32 bit or 16 bit targets.

### Advantage

- Detect obvious cases in which a program won't build for a different platform
- Detect incorrect uses of `usize`/`isize`, and suggest replacing them with `u64`/`i64`

### Example

```rust
// Example 1
const VALUE: usize = 5000000000;

// Example 2
fn check(val: usize) -> bool {
    val < 5000000000
}
```

Could be written as:

```rust
// Example 1
const VALUE: u64 = 5000000000;

// Example 2
fn check(val: u64) -> bool {
    val < 5000000000
}
```

**Comment by leonardo-m** [comment]
>Detect obvious cases in which a program won't build for a different platform<

This seems a nice idea. Do we have other similar potential-incompatibily situations that Clippy could help detect?
